### PR TITLE
Extract::Plugin::Haml.

### PR DESCRIPTION
**Pull request by calumhalcrow**
Hi again,

This is the plugin itself. I added related tests to t/5-extract.t. I updated the MANIFEST file but not META.yml, even though it should maybe recommend that Text::Haml be present. I wasn't sure what to put in the AUTHORS pod field so I just added my name underneath Audrey's.

Let me know if you have any questions etc.

Calum

### CODE REVIEW

**Comment by calumhalcrow** [comment]
Update: This now also includes changes as discussed on a [previous pull request](https://github.com/clintongormley/locale-maketext-lexicon/pull/7).

I implemented the following change in behaviour when no plugins are specified in xgettext.pl arguments:

```
if no plugins specified:
  select plugins based on file type.
  if no plugins match file type:
    use all plugins.
```

Or as stated in the changed documentation:

```
By default, if no plugins are specified, it first tries to determine which
plugins are intended specifically for the file type and uses them. If no
such plugins are found, it then uses all of the builtin plugins, overriding
the file types specified in each.
```

Let me know if there are any problems with these new changes and I will adjust accordingly.

Thanks,
Calum

File: lib/Locale/Maketext/Extract.pm [hunk 1]
```diff
@@ -265,13 +265,14 @@ our %Known_Plugins = (
...
                     formfu  => 'Locale::Maketext::Extract::Plugin::FormFu',
+                    haml    => 'Locale::Maketext::Extract::Plugin::Haml',
 );
 
 sub new {
     my $class   = shift;
     my %params  = @_;
     my $plugins = delete $params{plugins}
-        || { map { $_ => '*' } keys %Known_Plugins };
+        || { map { $_ => undef } keys %Known_Plugins };
```
Comment (calumhalcrow) [inline lib/Locale/Maketext/Extract.pm:275]: Previously, if an instance of Locale::Maketext::Extract was created without any arguments given to the constructor the list of available plugins stored on the `plugins` attribute were configured to act on any file (caused by `'*'` here). With this change (and the accompanying one on line 351), there will be no arguments passed to the plugins' constructor, so the plugins will be configured to act on only files they recognize (according to `known_file_type`). This behaviour is then used in the new `_plugins_specifically_for_file' method which you can see in the patch.

**Comment by clintongormley** [comment]
Hi Callum

Sorry this has taken so long.  Some of the tests in t/52 were failing because missing dependencies weren't taken into account, so it took me a while to get around to looking at this.

I've finally pushed v 0.93, which includes this change

thanks

Clint

**Comment by calumhalcrow** [comment]
Cool, thanks!

**Pull request closed without merging**